runtime/auto-split.sh

```diff
@@ -8,14 +8,7 @@ n_outputs="$#"
 # Set a default DISH_TOP in this directory if it doesn't exist
 PASH_TOP=${PASH_TOP:-$(git rev-parse --show-toplevel)}
 
-if type lsb_release >/dev/null 2>&1 ; then
-   distro=$(lsb_release -i -s)
-elif [ -e /etc/os-release ] ; then
-   distro=$(awk -F= '$1 == "ID" {print $2}' /etc/os-release)
-fi
-
-# convert to lowercase
-distro=$(printf '%s\n' "$distro" | LC_ALL=C tr '[:upper:]' '[:lower:]')
+distro=${1??Distro not given}
```

Comment:
This change requires that calls to `auto-split.sh` are also modified which would require some changes in the compiler. Maybe we can revert it for now and leave a todo or add an issue?

Reply:
Yeah you are correct, I thought I had addressed this issue.
I am switching all the `mktemp` commands to the freebsd way.
This approach however cannot be applied to all the commands (since the arguments and the flags differ) i.e., https://github.com/binpash/pash/blob/68c5065ef54263806d3bbb37c0ada6d005c76f05/runtime/wait_for_output_and_sigpipe_rest.sh#L15-L23